### ISSUE TITLE
Unable to get servicesInfo from nested configuration using Kotlin DSL 

### ISSUE THREAD

**Issue by rp199** [avast/gradle-docker-compose-plugin#289]
I'm using kotlin dsl with the latest version of the plugin and encountered a problem where I cannot access the serviceInfo from my nested configurations.

I have something like this:

```
dockerCompose {
    createNested("myNested").apply {
        //some custom config
    }
}
```
Not sure how can I translate the groovy snippet for getting the nested configuration:

Groovy:
```
test.doFirst {
    dockerCompose.myNested.exposeAsEnvironment(test)  
}
```

Already tried to invoke the `def methodMissing(String name, def args)` method like this:

```
dockerCompose.methodMissing("myNested", arrayOf<Any>())
```

But the it end ups trying to create the nested configuration instead of getting the existing one. 

Any idea how can I do this with Koltin DSL?

**Comment by augi** [comment]
Hello, as I have zero experience with Kotlin DSL, I'm not able to help you with this, sorry 😢

**Comment by rp199** [comment]
If I've understood correctly, the groovy way falls back to the last condition of the `methodMissing` method that calls `private ComposeSettings getOrCreateNested(String name)`. 

I'd expect that my approach would fall back to that condition too:
`dockerCompose.methodMissing("myNested", arrayOf<Any>())`

But either is failing into the second condition of the `methodMissing` or reaching the third one but is not able to find the nested condition for some reason and tries to create a new one with the same name.

Nevertheless, even if it works, I don't think it would be ideal for Kotlin DSL users, maybe we could expose a method for getting the nested configuration by name? Something like `getOrCreateNested(String name)` but without creating if not found. That would solve the problem for kotlin users

What do you think? I'm happy to try to create a PR myself with that change

**Comment by augi** [comment]
You are absolutely right! I've prepared #290 so feel free to comment it.

**Comment by augi** [comment]
The newly introduced `nested` method will be released in `0.14.3`.

**Comment by rp199** [comment]
Looks good! Thanks
